### ISSUE TITLE
introduce errors db

### ISSUE THREAD

**Issue by adecaro** [hyperledger-labs/fabric-token-sdk#714]
In cases when a transaction cannot be processed for some reasons, it would be good to have a database that stores information about these events. 
Such a database can be then consulted by a human or an agent to resolve the issues.
Here is an example, for some reasons, a failure happened during the update of the tokens db. The transaction remains in pending state and the tokens are not made available. This can cause misalignments that can be reflected in a wrong balance. 
The errors db can then be used to track these situations and resolve them timely.

**Comment by adecaro** [comment]
@arner , what do you think?

**Comment by arner** [comment]
Yes I like it. Especially (or only?) non-recoverable issues have to be persisted. A new table is a good place for that. By the way, sidestep, I wonder if we can be more granular in the status. Some 'Pending' transactions can technically still be committed (after endorsement) but some should be market 'Failed' right away.

**Comment by alexandrosfilios** [comment]
This should be implemented in FSC though IMHO
